index.js

```diff
@@ -2963,10 +2963,11 @@
 
   //. ### Array
 
-  //# append :: a -> Array a -> Array a
+  //# append :: (Applicative f, Semigroup (f a)) => a -> f a -> f a
   //.
-  //. Takes a value of any type and an array of values of that type, and
-  //. returns the result of appending the value to the array.
+  //. Takes a value of any type and a (Applicative, Semigroup) of values of
+  //. that type, and returns the result of appending the value to the
+  //. (Applicative, Semigroup).
```

Comment:
Perhaps a shorter description is more appropriate now that the function is so general. One option:

_Returns the result of appending its first argument to its second argument._

The constraints are well expressed in the type signature, and the examples should make the behaviour clear. On that note, let's add a non-array example below. :)